src/Input.js

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import { View, Text, TextInput, StyleSheet, Dimensions, TouchableOpacity } from 'react-native';
+import PropTypes from 'prop-types';
+import { Icon } from 'expo';
+
+// TO-DO:
+// 1. State functionality for Redux/Context/basic state stuff
+// 2. Maybe options for changind the View text for the viewPass button
+// 3. Idk. What else should we do in order to make this even more reusable.
+
+const { width } = Dimensions.get('window');
+
+const BASE_SIZE = 14;
+const COLOR_DEFAULT = `#293042`;
+const COLOR_BACKGROUND = `#FFFFFF`;
+const COLOR_BLACK = "#000";
+
+class Input extends React.Component {
+  static defaultProps = {
+    type: 'default',
+    password: false,
+    placeholderTextColor: COLOR_BLACK,
+    label: null,
+    help: null,
+    rounded: false,
+    left: true,
+    right: false,
+    viewPass: false
+  };
+
+  state = {
+    password: false
+  }
+
+  componentDidMount() {
+    this.setState({
+      password: this.props.password
+    })
+  }
+
+  render() {
+    const {
+      style,
+      type,
+      password,
+      placeholderTextColor,
+      label,
+      color,
+      help,
+      bgColor,
+      borderless,
+      borderColor,
+      viewPass,
+      rounded,
+      icon,
+      family,
+      left,
+      right,
+      ...props
+    } = this.props;
+    const inputViewStyles = [
+      styles.inputStyle,
+      styles.inputContainer,
+      bgColor && { backgroundColor: bgColor },
+      rounded && styles.rounded,
+      borderless && styles.borderless,
+      borderColor && { borderColor: borderColor},
+      style,
+    ];
+
+    const inputStyles = [
+      styles.inputView,
+      styles.inputIcon,
+      styles.inputText,
+      color && { color }
+    ];
+
+    const { [family]: IconFamily } = Icon;
```

Comment:
Can use the Icon component from https://github.com/galio-org/galio/blob/Alpha-1/src/Icon.js